lib/Class/Method/Modifiers.pm

```diff
@@ -218,6 +231,84 @@ sub _is_in_package {
     return $cv->GV->STASH->NAME eq $package;
 }
 
+# operator overloads are stored in the symbol table as "($op"
+#
+# if the overload is a coderef
+#    *{$symbol}{CODE} = $coderef
+#
+# if the overload is a $method_name
+#    *{$symbol}{CODE}   = \&overload::nil (or ::_nil)
+#    *{$symbol}{SCALAR} = $method_name
+#
+# cribbed from Role::Tiny
+
+# stolen from Role::Tiny
+sub _getglob {
+    no strict 'refs';
+    \*{ $_[0] };
+}
+
+sub _overloaded_op_method {
+
+    require( _MRO_MODULE );
+
+    my ( $class, $op ) = @_;
+
+    my $symbol = '(' . $op;
+
+    my $coderef;
+    my $method_name;
+
+    my $isa = mro::get_linear_isa( $class );
+
+    for my $pkg ( @$isa ) {
+
+        next
+          unless defined( $coderef = *{ _getglob "${pkg}::${symbol}" }{CODE} );
+
+        # method name ?
+        if (
+            ( defined &overload::nil && $coderef == \&overload::nil )
+            || ( defined &overload::_nil
+                && $coderef == \&overload::_nil ) )
+        {
+
+            $method_name = ${ *{ _getglob "${pkg}::${symbol}" }{SCALAR} };
+
+            # weird but possible?
+            return unless defined $method_name;
+        }
+
+        last;
+    }
+
+    return if !defined $coderef;
+
+    # if $method_name is undefined, then $coderef is the real method
+    # to call. we can only modify named subroutines, so create a slot
+    # in the package's symbol table and rewire the overload.
+
+    if ( !defined $method_name ) {
+
+        $method_name = $class . $symbol;
+
+        # stolen from Sub::Quote::_sanitize_identifier
+
+        $method_name =~ s/([_\W])/sprintf('_%x', ord($1))/ge;
+
+        *{ _getglob "${class}::${method_name}" } = $coderef;
+
+        my $glob = _getglob "${class}::${symbol}";
+        *$glob = \$method_name;
+        no warnings 'redefine';
+        *$glob = \&overload::nil || \&overload::_nil;
```

Comment:
There was an idea to change `nil` to be `_nil` at some point that never went anywhere afaik.  We may want to just forget about supporting that.  If we are supporting that though, this won't work.  `\&overload::nil` will force the code-ref to exist and will never be false.  It needs a defined check like is used above.